firewood/src/merkle.rs

```diff
@@ -390,6 +394,13 @@ impl<T: HashedNodeReader> Merkle<T> {
                         None => continue,
                         Some(Child::Node(_)) => continue, // TODO
                         Some(Child::AddressWithHash(addr, hash)) => (*addr, Some(hash)),
+                        Some(Child::MaybePersisted(maybe_persisted, hash)) => {
+                            // For MaybePersisted, we need the address if it's persisted
```

Comment:
Same, this redundant. Feels like we need a method to yield `Result<(LinearAddress, HashType, _>` on child.